src/main/java/net/fabricmc/loom/configuration/InstallerData.java

```diff
@@ -45,35 +53,61 @@ public void applyToProject(Project project) {
 
 		extension.setInstallerData(this);
 
-		JsonObject libraries = installerJson.get("libraries").getAsJsonObject();
+		final JsonObject libraries = installerJson.get("libraries").getAsJsonObject();
+
+		applyDependendencies(libraries.get("common").getAsJsonArray(), project);
+
+		// Apply development dependencies if they exist.
+		if (libraries.has("development")) {
+			applyDependendencies(libraries.get("development").getAsJsonArray(), project);
+		}
+	}
+
+	private void applyDependendencies(JsonArray jsonArray, Project project) {
+		LoomGradleExtension extension = LoomGradleExtension.get(project);
 		Configuration loaderDepsConfig = project.getConfigurations().getByName(Constants.Configurations.LOADER_DEPENDENCIES);
-		Configuration apDepsConfig = project.getConfigurations().getByName("annotationProcessor");
+		Configuration annotationProcessor = project.getConfigurations().getByName(JavaPlugin.ANNOTATION_PROCESSOR_CONFIGURATION_NAME);
+
+		for (JsonElement jsonElement : jsonArray) {
+			final JsonObject jsonObject = jsonElement.getAsJsonObject();
+			final String name = jsonObject.get("name").getAsString();
 
-		libraries.get("common").getAsJsonArray().forEach(jsonElement -> {
-			String name = jsonElement.getAsJsonObject().get("name").getAsString();
-			project.getLogger().debug("Adding dependency ({}) from installer JSON", name);
+			LOGGER.debug("Adding dependency ({}) from installer JSON", name);
 
 			ExternalModuleDependency modDep = (ExternalModuleDependency) project.getDependencies().create(name);
-			modDep.setTransitive(false);
+			modDep.setTransitive(false); // Match the launcher in not being transitive
 			loaderDepsConfig.getDependencies().add(modDep);
 
-			// TODO: work around until https://github.com/FabricMC/Mixin/pull/60 and https://github.com/FabricMC/fabric-mixin-compile-extensions/issues/14 is fixed.
+			// Work around https://github.com/FabricMC/Mixin/pull/60 and https://github.com/FabricMC/fabric-mixin-compile-extensions/issues/14 is fixed.
```

Comment:
```suggestion
			// Work around until https://github.com/FabricMC/Mixin/pull/60 and https://github.com/FabricMC/fabric-mixin-compile-extensions/issues/14 is fixed.
```

Reply:
I dont think we can ever remove this tbh, it makes sense to keep.

Reply:
Read the whole comment until the end 😅

Reply:
Good idea :D Should be better now 👍